lib/cache.js

```diff
@@ -32,12 +33,26 @@ class MemoryCache {
    * @return {void}
    */
   set (key, value, ttl) {
+    // Evict oldest entry when at capacity.
+    if (this.cache.size === this.maxEntries) {
+      this.cache.delete(this.cache.keys().next().value);
+    }
+
     this.cache.set(key, {
       expiresAt: ttl,
       value
     });
   }
 
+  /**
+   * Deletes cache entry for key
+   * @param {string} key Cache key
+   * @return {void}
+   */
+  delete (key) {
```

Comment:
i think creating this function to only use once is a unnecessary

Reply:
Fair enough. I added it because I thought about removing coalesce entries after resolving (so no overlap between cache types), but considered that might be a bit too much - it now relies on 'normal' eviction instead.

I'll remove it.